core/src/main/java/org/apache/carbondata/core/scan/result/iterator/RawResultIterator.java

```diff
@@ -66,18 +81,62 @@
 
   public RawResultIterator(CarbonIterator<RowBatch> detailRawQueryResultIterator,
       SegmentProperties sourceSegProperties, SegmentProperties destinationSegProperties,
-      boolean isStreamingHandoff) {
+      boolean isStreamingHandoff, boolean hasColumnDrift) {
```

Comment:
If possible we can create new class and extend this class and override the convertRow method for better readability